src/coreclr/jit/optimizebools.cpp

```diff
@@ -2022,8 +1663,134 @@ PhaseStatus Compiler::optOptimizeBools()
+    {
+        return modified;
+    }
+
+    // Both edges must be BBJ_RETURN
+    BasicBlock* retFalseBb = block->GetFalseTarget();
+    BasicBlock* retTrueBb  = block->GetTrueTarget();
+
+    // Although, we might want to fold fallthrough BBJ_ALWAYS blocks first
+    if (fgCanCompactBlock(retTrueBb))
+    {
+        fgCompactBlock(retTrueBb);
+        modified = true;
+    }
+    // By the time we get to the retFalseBb, it might be removed by fgCompactBlock()
+    // so we need to check if it is still valid.
+    if (!retFalseBb->HasFlag(BBF_REMOVED) && fgCanCompactBlock(retFalseBb))
+    {
+        fgCompactBlock(retFalseBb);
+        modified = true;
+    }
+    // Same here - bail out if the block is no longer BBJ_COND after compacting.
+    if (!block->KindIs(BBJ_COND))
+    {
+        return modified;
+    }
+
+    retTrueBb  = block->GetTrueTarget();
+    retFalseBb = block->GetFalseTarget();
+    if (!retTrueBb->KindIs(BBJ_RETURN) || !retFalseBb->KindIs(BBJ_RETURN) ||
+        !BasicBlock::sameEHRegion(block, retTrueBb) || !BasicBlock::sameEHRegion(block, retFalseBb))
+    {
+        // Both edges must be BBJ_RETURN
+        return modified;
+    }
+
+    // The last statement has to be either JTRUE(cond) or JTRUE(comma(cond)),
+    // but let's be resilient just in case.
+    assert(block->lastStmt() != nullptr);
+    GenTree* node = block->lastStmt()->GetRootNode();
+    GenTree* cond = node->gtGetOp1();
+    if (!cond->OperIsCompare())
+    {
+        return modified;
+    }
+    assert(cond->TypeIs(TYP_INT));
+
+    if ((retTrueBb->GetUniquePred(this) == nullptr) && (retFalseBb->GetUniquePred(this) == nullptr))
+    {
+        // Both return blocks have multiple predecessors - bail out.
+        // We don't want to introduce a new epilogue.
+        return modified;
+    }
+
+    // Is block a BBJ_RETURN(1/0) ? (single statement)
+    auto isReturnBool = [](const BasicBlock* block, bool value) {
+        if (block->KindIs(BBJ_RETURN) && block->hasSingleStmt() && (block->lastStmt() != nullptr))
+        {
+            GenTree* node = block->lastStmt()->GetRootNode();
+            return node->OperIs(GT_RETURN) && node->gtGetOp1()->IsIntegralConst(value ? 1 : 0);
+        }
+        return false;
+    };
+
+    // Make sure we deal with true/false return blocks (or false/true)
+    bool retTrueFalse = isReturnBool(retTrueBb, true) && isReturnBool(retFalseBb, false);
+    bool retFalseTrue = isReturnBool(retTrueBb, false) && isReturnBool(retFalseBb, true);
+    if (!retTrueFalse && !retFalseTrue)
+    {
+        return modified;
+    }
+
+    // Reverse the condition if we jump to "return false" on true.
+    if (retFalseTrue)
+    {
+        gtReverseCond(cond);
+    }
+    modified = true;
+
+    // Decrease the weight of the return blocks since we no longer have edges to them.
+    // Although, they still might be reachable from other blocks (at least one of them).
```

Comment:
```suggestion
    // Although one might still be reachable from other blocks.
```

Sorry one more nit

Reply:
> Sorry one more nit

Thanks! I'm more than happy to see people rewording my bad English 🙂